client/src/store/toolStore.js

```diff
@@ -57,13 +32,31 @@ const actions = {
         const { data } = await axios.get(`${getAppRoot()}api/tools/${toolId}`);
         commit("saveToolForId", { toolId, toolData: data });
     },
-    fetchAllTools: async ({ state, commit }) => {
-        // Preventing store from being populated for every search: we fetch again only if:
-        // store isn't already populated (initial fetch)
-        if (state.toolsList.length === 0) {
-            console.log("fetching all tools once");
-            const { data } = await axios.get(`${getAppRoot()}api/tools?in_panel=False`);
-            commit("saveTools", { toolsData: data });
+    fetchAllTools: async ({ commit }, { filterSettings }) => {
+        if (Object.keys(filterSettings).length !== 0) {
+            // Parsing filterSettings to Whoosh query
+            let q = "";
+            for (const [key, filterValue] of Object.entries(filterSettings)) {
```

Comment:
Suggesting to push this into its own utility file and to add a test-function like e.g. here https://github.com/galaxyproject/galaxy/blob/dev/client/src/utils/url.test.js.